storage/innobase/srv/srv0srv.cc

```diff
@@ -1625,6 +1682,32 @@ srv_export_innodb_status(void)
 	}
 
 	mutex_exit(&srv_innodb_monitor_mutex);
+
+	export_vars.innodb_ibuf_discarded_delete_marks = 
+		ibuf->n_discarded_ops[IBUF_OP_DELETE_MARK];
+	export_vars.innodb_ibuf_discarded_deletes =
+		ibuf->n_discarded_ops[IBUF_OP_DELETE];
+	export_vars.innodb_ibuf_discarded_inserts =
+		ibuf->n_merged_ops[IBUF_OP_INSERT];
+	export_vars.innodb_ibuf_free_list = ibuf->free_list_len;
+	export_vars.innodb_ibuf_merged_delete_marks = 
+		ibuf->n_merged_ops[IBUF_OP_DELETE_MARK];
+	export_vars.innodb_ibuf_merged_deletes = ibuf->n_merged_ops[IBUF_OP_DELETE];
+	export_vars.innodb_ibuf_merged_inserts = ibuf->n_merged_ops[IBUF_OP_INSERT];
+	export_vars.innodb_ibuf_merges = ibuf->n_merges;
+	export_vars.innodb_ibuf_segment_size = ibuf->seg_size;
+	export_vars.innodb_ibuf_size = ibuf->size;
+
+	mutex_enter(&log_sys.mutex);
+
+	export_vars.innodb_lsn_current = log_sys.lsn;
+	export_vars.innodb_lsn_flushed = log_sys.flushed_to_disk_lsn;
+	export_vars.innodb_lsn_last_checkpoint = log_sys.last_checkpoint_lsn;
+	export_vars.innodb_checkpoint_age = (log_sys.lsn - 
+		log_sys.last_checkpoint_lsn);
+	export_vars.innodb_checkpoint_max_age = log_sys.max_checkpoint_age;
+
+	mutex_exit(&log_sys.mutex);
```

Comment:
These are protected by mutex, and we probably want to deliver a consistent snapshot of these variables. Therefore, we must keep this code like it is now.